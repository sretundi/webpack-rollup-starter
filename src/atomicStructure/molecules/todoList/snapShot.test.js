
import React from 'react';
import Renderer from 'react-test-renderer';

import TodoList from './todoList.jsx';

// with mocks in place in renders child components as react components
// without the mocks it renders the html elements within the child components
// jest.mock('../../atoms/inputs/radioInput.jsx', () => 'Radio Input');
// jest.mock('../../atoms/inputs/textInput.jsx', () => 'text input');
// jest.mock('../../atoms/buttons/buttonCommon.jsx', () => 'button common');

it('renders todo list correctly', () => {
    const tree = Renderer
        .create(
            <TodoList
                items={[1, 2]}
                editableItem={{isEditing: false}}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
