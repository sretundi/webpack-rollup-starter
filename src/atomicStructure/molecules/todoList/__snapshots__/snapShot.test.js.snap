// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders todo list correctly 1`] = `
<ul
  className="dualButtonListContainer"
>
  <li
    className="listItemContainer"
  >
    <input
      type="radio"
    />
    <input
      className="b-textInput--removeBorder b-textInput--removeOutline b-textInput"
      onChange={[Function]}
      type="text"
      value={1}
    />
    <div
      className="listItemButtonContainer"
    >
      <button
        className=" smButton b-button"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className=" listItemDeleteBtn smButton b-button"
        onClick={[Function]}
      >
        delete
      </button>
    </div>
  </li>
  <li
    className="listItemContainer"
  >
    <input
      type="radio"
    />
    <input
      className="b-textInput--removeBorder b-textInput--removeOutline b-textInput"
      onChange={[Function]}
      type="text"
      value={2}
    />
    <div
      className="listItemButtonContainer"
    >
      <button
        className=" smButton b-button"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className=" listItemDeleteBtn smButton b-button"
        onClick={[Function]}
      >
        delete
      </button>
    </div>
  </li>
</ul>
`;
